dataland-frontend/src/utils/FileUploadUtils.ts

```diff
@@ -0,0 +1,86 @@
+import { CompanyReport } from "@clients/backend";
+import { ApiClientProvider } from "@/services/ApiClients";
+import Keycloak from "keycloak-js";
+
+export interface DocumentToUpload {
+  file: File;
+  fileNameWithoutSuffix: string;
+
+  reference: string;
+}
+
+export interface StoredReport extends CompanyReport {
+  reportName: string;
+}
+
+export interface ReportToUpload extends CompanyReport {
+  file: File;
+  fileNameWithoutSuffix: string;
+}
+
+/**
+ * uploads Files through the frontend
+ * @param files the list of files to upload
+ * @param getKeycloakPromise getter for a keycloak promise
+ */
+export async function uploadFiles(
+  files: ReportToUpload[] | DocumentToUpload[],
+  getKeycloakPromise: () => Promise<Keycloak>
+): Promise<void> {
+  const documentControllerApi = await new ApiClientProvider(getKeycloakPromise()).getDocumentControllerApi();
+  for (const fileToUpload of files) {
+    const fileIsAlreadyInStorage = (await documentControllerApi.checkDocument(fileToUpload.reference)).data
+      .documentExists;
+    if (!fileIsAlreadyInStorage) {
+      const backendComputedHash = (await documentControllerApi.postDocument(fileToUpload.file)).data.documentId;
+      if (fileToUpload.reference !== backendComputedHash) {
+        throw Error("Locally computed document hash does not concede with the one received by the upload request!");
+      }
+    }
+  }
+}
+
+/**
+ * Checks if there was actually a file added by the user that was not filtered
+ * out by the FileUpload component.
+ * @param filesCurrentlySelectedByUser the files currently selected by the user
+ * @param previouslySelectedCertificates the reports that have already been selected before the last change
+ * @returns true if there is actually a file added by the user
+ */
+export function isThereActuallyANewFileSelected(
```

Comment:
Why is this method necessary?

Reply:
This was implemented in the "upload via UI EU Taxonomy" story. So I don't know, but we have kept to not destroy anything.

Reply:
ok, then we keep it as it is